src/parser/TurtleParser.cpp

```diff
@@ -0,0 +1,409 @@
+// Copyright 2018, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Kalmbach(joka921) <johannes.kalmbach@gmail.com>
+//
+
+#include "./TurtleParser.h"
+#include <string.h>
+
+// TODO: always disambiguate if we indeed take the longest match
+
+// _______________________________________________________________
+bool TurtleParser::statement() {
+  _tok.skipWhitespaceAndComments();
+  return directive() || (triples() && skip(_tokens.Dot));
+}
+
+// ______________________________________________________________
+bool TurtleParser::directive() {
+  return prefixID() || base() || sparqlPrefix() || sparqlBase();
+}
+
+// ________________________________________________________________
+bool TurtleParser::prefixID() {
+  if (skip(_tokens.TurtlePrefix)) {
+    if (check(pnameNS()) && check(iriref()) && check(skip(_tokens.Dot))) {
+      // strip  the angled brackes <bla> -> bla
+      _prefixMap[_activePrefix] =
+          _lastParseResult.substr(1, _lastParseResult.size() - 2);
```

Comment:
add comment why `-2`, from the above comment I'd think it would be `-1` but I guess it's for the `.` at the end